app/src/main/java/net/gsantner/opoc/util/CoolExperimentalStuff.java

```diff
@@ -141,4 +143,44 @@ public void run() {
             w.setLayout(WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.MATCH_PARENT);
         }
     }
+
+    private static Thread autoScrollerThread = null;
+
+    public static void startAutoScroller(final Activity activity, final boolean isWebView) {
+        View view;
+        if (isWebView) {
+            view = activity.findViewById(R.id.document__fragment_view_webview);
+        } else {
+            view = activity.findViewById(R.id.document__fragment__edit__content_editor__scrolling_parent);
+        }
+
+        final int scrollAmount = view.getHeight() / 3;
+        final int millisecondsDelay = 5000;
+
+        autoScrollerThread = new Thread() {
+            @Override
+            public void run() {
+                try {
+                    while (view.getVisibility() == View.VISIBLE) {
+                        Thread.sleep(millisecondsDelay);
+                        view.scrollBy(0, scrollAmount);
```

Comment:
doeys that workgood and same for edittrxt and webview?

(i dont kmow, question)

Reply:
Oh I believe the edit view never goes to another state than View.VISIBLE, this check only works on webview, as the edit view never switches to View.GONE. I think this while loop can be converted into a while (true), as the thread is set to be always killed when setDocumentViewVisibility is invoked.

Reply:
I could convert it to kill the thread when it scrolls to the bottom, too